src/utils/IntlWrapper.ts

```diff
@@ -0,0 +1,376 @@
+  months: 30 * 24 * 60 * 60 * 1000,
+  weeks: 7 * 24 * 60 * 60 * 1000,
+  days: 24 * 60 * 60 * 1000,
+  hours: 60 * 60 * 1000,
+  minutes: 60 * 1000,
+  seconds: 1000,
+  milliseconds: 1,
+} as const;
+
+export class IntlDuration {
+  protected duration?: durationFormat;
+
+  protected locale: Intl.LocalesArgument;
+
+  // @ts-expect-error surely it works
+  protected isFallback = !Intl.DurationFormat;
+
+  constructor(locale: Intl.LocalesArgument = api.storage.lang('locale')) {
+    this.locale = locale;
+    return this;
+  }
+
+  setLocale(locale: Intl.LocalesArgument) {
+    this.locale = locale;
+    return this;
+  }
+
+  // For {hours: 2, minutes: 15}
+  setDuration(duration: durationFormat) {
+    this.duration = duration;
+    return this;
+  }
+
+  // For {minutes: 155}
+  setDurationFormatted(duration: durationFormat, style: durationStyle = 'Duration') {
+    const ms = IntlDuration.durationToMs(duration);
+    this.setRelativeTime(ms, 'milliseconds', style);
+    return this;
+  }
+
+  // For 9000 (seconds)
+  setRelativeTime(
+    relativeTime: number,
+    convertFrom: keyof typeof dateUnitToMs,
+    style: durationStyle = 'Duration',
+  ) {
+    const ms = relativeTime * dateUnitToMs[convertFrom];
+    this.duration = this.process(ms, style);
+    return this;
+  }
+
+  // For 1733320467580 (timestamp)
+  setTimestamp(
+    timestamp: number,
+    style: durationStyle = 'Duration',
+    relativeTo: Date | number = new Date(),
+  ) {
+    const ms = timestamp - new Date(relativeTo).getTime();
+    this.setRelativeTime(ms, 'milliseconds', style);
+    return this;
+  }
+
+  protected process(relativeTime: number, style: durationStyle) {
+    switch (style) {
+      case 'Duration':
+        return relativeToDuration(relativeTime, ['seconds', 'minutes', 'hours', 'days']).duration;
+      case 'Progress':
+        if (!relativeTime) break;
+        const time = relativeToDuration(relativeTime, [
+          'seconds',
+          'minutes',
+          'hours',
+          'days',
+          'years',
+        ]);
+        return shortTime(time.duration);
+      case 'M/H/D/Y':
+        if (!relativeTime) break;
+        return relativeToDuration(relativeTime, ['minutes', 'hours', 'days', 'years']).duration;
+      default:
+        return {};
+    }
+    return {};
+  }
+
+  static durationToMs(input: durationFormat): number {
+    const keys = Object.keys(input);
+    let timestamp = 0;
+    if (keys.length <= 0) return timestamp;
+    for (let i = 0; i < keys.length; i++) {
+      const key = keys[i];
+      const value = input[key];
+      timestamp += dateUnitToMs[key] * value;
+    }
+    return timestamp;
+  }
+
+  getRelativeText(format: durationFormatStyle = 'narrow'): string {
+    if (!this.duration) return '';
+    if (this.isFallback) return timeToString(this.duration);
+    // @ts-expect-error surely it works
+    return new Intl.DurationFormat(this.locale, { style: format }).format(this.duration);
```

Comment:
I am not fully happy with output for the narrow progress outputs. Long does not look good either.

I tested around using narrow but still having the "most significant bit/dateunit" as long. It seemed a good compromise between narrow and long. I looked at english, spanish and german and it looked fine. What do you think? 
![image](https://github.com/user-attachments/assets/b182b3b2-9fb4-490d-83fe-3ab01b5b9345)
![image](https://github.com/user-attachments/assets/09db13f1-8299-4370-ab89-478a35cbf49b)
![image](https://github.com/user-attachments/assets/10873f14-f2ce-4019-b7f4-0c119426bd9f)
![image](https://github.com/user-attachments/assets/d0db42e4-1d01-42ed-ad7e-23d21bd7c3e3)
![image](https://github.com/user-attachments/assets/9cbb0d55-0fd1-4fca-9a09-2bd2898406d1)


Examples:
 `{days: 2, hours: 45}` -> `{ days: 'long', style: 'narrow'}`
`{hours: 2, minutes 45}` -> `{ hours: 'long', style: 'narrow'}`
etc...

`Intl.DurationFormat(this.locale, { days: 'long', style: 'narrow'}).format(this.duration);`

Implementation wise I think the best way is to add an additional custom durationFormatStyle(`narrowMSB/primaryLongOthersNarrow/significantLongNarrow`) and then handle it in getRelativeText.

Reply:
I assume you don't like "short"? 
Well I like how your screenshots look like, so why not. Butto be honest I have no clue what `narrowMSB` means and what diff between other 2 lmao
I guess `significantLongNarrow` is the first date unit is "long" format and any other is "narrow"?